Neos.Flow/Classes/Security/Exception/AuthenticationRequiredException.php

```diff
@@ -22,4 +24,37 @@ class AuthenticationRequiredException extends \Neos\Flow\Security\Exception
      * @var integer
      */
     protected $statusCode = 401;
+
+    /**
+     * @var ActionRequest
+     */
+    protected $interceptedRequest;
+
+    /**
+     * Attach the given action request as intercepted request and return self.
+     *
+     * @param ActionRequest $actionRequest
+     * @return AuthenticationRequiredException
+     */
+    public function withInterceptedRequest(ActionRequest $actionRequest): self
+    {
+        $this->interceptedRequest = $actionRequest;
```

Comment:
Why not clone $this and make it immutable? Because exceptions can not be cloned in PHP :)

Reply:
Good point, I wasn't aware of that.
Just a minor issue with this. The `with*` pattern is used for immutable objects in our code base (and commonly in PHP anyways). I think we should go for a good old setter in this case, if immutability is not an option